src/Workspaces/MSBuildTest/VisualStudioMSBuildWorkspaceTests.cs

```diff
@@ -0,0 +1,219 @@
+            Assert.Equal(5, "//\u00E2\u20AC\u0153".Length);
+            Assert.Equal("//\u00E2\u20AC\u0153".Length, text.Length);
+        }
+
+        [ConditionalFact(typeof(VisualStudioMSBuildInstalled)), Trait(Traits.Feature, Traits.Features.MSBuildWorkspace)]
+        public void TestOpenProject_WithInvalidFilePath_Fails()
+        {
+            CreateFiles(GetSimpleCSharpSolutionFiles());
+            var projectFilePath = GetSolutionFileName(@"http://localhost/Invalid/InvalidProject.csproj");
+
+            using var workspace = CreateMSBuildWorkspace();
+
+            AssertEx.Throws<InvalidOperationException>(() => workspace.OpenProjectAsync(projectFilePath).Wait());
+        }
+
+        [ConditionalFact(typeof(VisualStudioMSBuildInstalled)), Trait(Traits.Feature, Traits.Features.MSBuildWorkspace)]
+        public void TestOpenProject_WithInvalidProjectReference_SkipFalse_Fails()
+        {
+            CreateFiles(GetMultiProjectSolutionFiles()
+                .WithFile(@"VisualBasicProject\VisualBasicProject.vbproj", Resources.ProjectFiles.VisualBasic.InvalidProjectReference));
+            var projectFilePath = GetSolutionFileName(@"VisualBasicProject\VisualBasicProject.vbproj");
+
+            using var workspace = CreateMSBuildWorkspace();
+            workspace.SkipUnrecognizedProjects = false;
+
+            AssertEx.Throws<InvalidOperationException>(() => workspace.OpenProjectAsync(projectFilePath).Wait());
+        }
+
+        [ConditionalFact(typeof(VisualStudioMSBuildInstalled)), Trait(Traits.Feature, Traits.Features.MSBuildWorkspace)]
+        public void TestOpenSolution_WithInvalidProjectPath_SkipFalse_Fails()
+        {
+            // when not skipped we should get an exception for the invalid project
+
+            CreateFiles(GetSimpleCSharpSolutionFiles()
+                .WithFile(@"TestSolution.sln", Resources.SolutionFiles.InvalidProjectPath));
+            var solutionFilePath = GetSolutionFileName(@"TestSolution.sln");
+
+            using var workspace = CreateMSBuildWorkspace();
+            workspace.SkipUnrecognizedProjects = false;
+
+            AssertEx.Throws<InvalidOperationException>(() => workspace.OpenSolutionAsync(solutionFilePath).Wait());
+        }
+
+        [ConditionalFact(typeof(VisualStudioMSBuildInstalled)), Trait(Traits.Feature, Traits.Features.MSBuildWorkspace)]
+        public void TestOpenSolution_WithInvalidSolutionFile_Fails()
+        {
+            CreateFiles(GetSimpleCSharpSolutionFiles());
+            var solutionFilePath = GetSolutionFileName(@"http://localhost/Invalid/InvalidSolution.sln");
+
+            using var workspace = CreateMSBuildWorkspace();
+
+            AssertEx.Throws<InvalidOperationException>(() => workspace.OpenSolutionAsync(solutionFilePath).Wait());
+        }
+
+        [ConditionalFact(typeof(VisualStudioMSBuildInstalled)), Trait(Traits.Feature, Traits.Features.MSBuildWorkspace)]
+        public async Task TestAddRemoveMetadataReference_GAC()
```

Comment:
No GAC on .NET Core

Reply:
![And There Was Much Rejoicing](https://user-images.githubusercontent.com/201340/112700855-520a5680-8e4c-11eb-9c2c-e2c365c21e14.gif)